### ISSUE TITLE
404.php No longer shows featured products

### ISSUE THREAD

**Issue by Trebacz** [woocommerce/storefront#608]
Pretty sure the issue is related to the change in WooCommerce 3.0 https://github.com/woocommerce/woocommerce/wiki/2.6.x-to-3.0.0-Developer-Migration-Notes#product-visibility-is-taxonomy-based-instead-of-meta-based

I have two products marked as featured in the admin interface and the 404 page always shows only sale items. In my limited troubleshooting it appears to get nothing back from the storefront_promoted_products(); call in 404.php

I can't figure out what should be changed in /inc/woocommerce/storefront-woocommerce-template-functions.php or in the woocommerce /includes/wc-product-functions.php...

**Comment by tiagonoronha** [comment]
@Trebacz the 404 page uses the WooCommerce Featured Products shortcode. I've tested it and it's working correctly.

If you recently upgraded to WooCommerce 3.0, you'll need to go to one of your featured products and save it again in order for the correct products show again.

**Comment by Trebacz** [comment]
Thanks for the suggestion.

I did upgrade from 2.6. I tried what you suggested. I also tried it with all plugins disabled, storefront them only and tried the project widget (to see if there was something different about the 404 page request). I tried making it as featured form the product list, quickedit and single product view. None of them triggered a change.

Perhaps it a problem unique to my implementation not triggering the Feature Products upgrade. I'll dig further....

**Comment by Trebacz** [comment]
@tiagonoronha, I'm posting here, just because I eventually found the source of my problem. "Somehow", I ended up with 2 entries in the wp_terms table in my wordpress database. I had some time to look into it and due to the issues scarcity, I think it's an edge case.

`SELECT * FROM wp_terms where name = 'featured';`
The only difference in each entry was the "term_id" and "slug" fields.

`SELECT * FROM wp_term_taxonomy where term_id in('87', '88');`
Looking at wp_term_taxonomy table, both were part of the "product_visibility' taxonomy in the wp_term_taxonomy. In the wp_term_taxonomy table one (term_id = 87) was correctly recording the count of feature products on my site. The second was always zero.

I'm guessing this duplicate entry is an artifact for a start and stop WC 2.6 to 3.0 upgrade (or something close to that). I did not have a clean upgrade to 3.0. We did intial do a database restore to get back to 2.6.

`'SELECT * FROM wp_term_relationships where term_taxonomy_id in ('87','88');'`
There are no posts related to this term_id in wp_term_relationships.

**FIX**: I updated the second entry in wp_terms "name" field from "featured" to "old-featured" in My SQL. After that our "featured products" functionality works as expected.

I don't like going straight to the database to fix things, but it seems to have solved my issues.
